app/controllers/api/account.php

```diff
@@ -49,20 +49,26 @@
     ->param('email', '', new Email(), 'User email.')
     ->param('password', '', new Password(), 'User password. Must be at least 8 chars.')
     ->param('name', '', new Text(128), 'User name. Max length: 128 chars.', true)
+    ->param('hash', 'bcrypt', new WhiteList(['bcrypt', 'scrypt', 'md5']), 'Hashing algorithm for password. The default value is bcrypt.', true)
+    ->param('import', false, new Boolean(), 'Are you importing hashed password?', true)
```

Comment:
```suggestion
    ->param('hash', 'bcrypt', new WhiteList(['bcrypt', 'scrypt', 'md5']), 'Hashing algorithm for password. The default value is bcrypt.', true)
    ->param('import', false, new Boolean(), 'Are you importing hashed password?', true)
```

My original idea was for the `hash` param to indicate what kind of string was passed as the password. `plain` should have been the default type, meaning the password was passed with no hashing. In this case we should hash it with our default algorithm.

Using this implementation we don't need the `import` param, and we don't need to change the default behavior of the API.

We can also use Auth::HASH_BCRYPT and other constants for supported algorithms.

Reply:
Also, changes should apply only to the users API, for server side import of users.

Reply:
We also need to add plaintext option